test/sharness/t0100-name.sh

```diff
@@ -8,209 +8,272 @@ test_description="Test ipfs repo operations"
+        printf '' | test_expect_code 1 ipfs name publish -Q --allow-offline  >publish_out 2>&1
+        '
+
+        test_expect_success "publish output has the correct error" '
+        grep "argument \"ipfs-path\" is required" publish_out
+        '
+
+        test_expect_success "'ipfs name publish --help' succeeds" '
+        ipfs name publish --help
+        '
+
+        # test offline resolve
+
+        test_expect_success "'ipfs name resolve --offline' succeeds" '
+        ipfs name resolve --offline "$PEERID" >output
+        '
+        test_expect_success "resolve output looks good" '
+        printf "/ipld/%s/thing\n" "$OBJECT_HASH" >expected4 &&
+        test_cmp expected4 output
+        '
+
+        test_expect_success "'ipfs name resolve --offline -n' succeeds" '
+        ipfs name resolve --offline -n "$PEERID" >output
+        '
+        test_expect_success "resolve output looks good" '
+        printf "/ipld/%s/thing\n" "$OBJECT_HASH" >expected4 &&
+        test_cmp expected4 output
+        '
+
+        test_launch_ipfs_daemon
+
+        test_expect_success "'ipfs name resolve --offline' succeeds" '
+        ipfs name resolve --offline "$PEERID" >output
+        '
+        test_expect_success "resolve output looks good" '
+        printf "/ipld/%s/thing\n" "$OBJECT_HASH" >expected4 &&
+        test_cmp expected4 output
+        '
+
+        test_expect_success "'ipfs name resolve --offline -n' succeeds" '
+        ipfs name resolve --offline -n "$PEERID" >output
+        '
+        test_expect_success "resolve output looks good" '
+        printf "/ipld/%s/thing\n" "$OBJECT_HASH" >expected4 &&
+        test_cmp expected4 output
+        '
+
+        test_expect_success "empty request to name publish doesn't panic and returns error" '
+        curl -X POST "http://$API_ADDR/api/v0/name/publish" > curl_out || true &&
+        grep "argument \"ipfs-path\" is required" curl_out
+        '
+
+        test_kill_ipfs_daemon
+
+        # Test daemon in offline mode
+        test_launch_ipfs_daemon --offline
+
+        test_expect_success "'ipfs name publish' fails offline mode" '
+        test_expect_code 1 ipfs name publish "/ipfs/$HASH_WELCOME_DOCS"
+        '
+
+        test_kill_ipfs_daemon
+
+        test_expect_success "clean up ipfs dir" '
+        rm -rf "$IPFS_PATH"
+        '
+}
+test_name_with_self 'rsa'
+test_name_with_self 'ed25519'
+
+test_name_with_key() {
+        GEN_ALG=$1
+
+        test_expect_success "ipfs init (key variant $GEN_ALG)" '
+        export IPFS_PATH="$(pwd)/.ipfs" &&
+        ipfs init --profile=test > /dev/null
+        '
+
+        test_expect_success "'prepare keys" '
+        case $GEN_ALG in
+        rsa)
+                export KEY=`ipfs key gen -f=b58mh --type=rsa --size=2048 key` &&
+                export KEY_B36CID=`ipfs key list -f=b36cid -l | grep key | cut -d " " -f1`
+                ;;
+        ed25519)
+                export KEY=`ipfs key gen -f=b36cid --type=ed25519 key`
```

Comment:
It seems like this should be exported as b58, same as for RSA. If we want to increase test coverage by testing that publishing with b58 results in b36 output and publishing with b36 results in b36 output we can just do that.

Reply:
this seems fine now